### ISSUE TITLE
Gnome 45 support

### ISSUE THREAD

**Issue by Koitharu** [Rafostar/gnome-shell-extension-pip-on-top#17]
Looks like this extension is not compatible with Gnome 45

```
SyntaxError: import declarations may only appear at top level of a module

Stack trace:
  @file:///usr/share/gnome-shell/extensions/pip-on-top@rafostar.github.com/prefs.js:2:24
  _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
  ```
  
https://gjs.guide/extensions/upgrading/gnome-shell-45.html#esm

**Comment by Scotty-Trees** [comment]
I love using this extension so thanks for making it. Now the next Gnome 45 release is going to change up some dynamics with extensions, so unless you update your extension to support ESM, it is **NOT** going to work on Gnome 45. There is a blog post from the Gnome devs here that details briefly the reasons for this change and what we can do to help update extensions to the new release: https://blogs.gnome.org/shell-dev/2023/09/02/extensions-in-gnome-45/

tl;dr, GNOME Shell moved from GJS’ own custom imports system to standard JavaScript modules (ESM), which breaks every extension unless it's updated to support Gnome 45. Please let us know if your extension can be updated to support Gnome 45 (it's release should be within the month btw). I hope this can be updated in time, and thanks again for making a very useful little extension for Firefox :+1:

**Comment by mbeps** [comment]
> I love using this extension so thanks for making it. Now the next Gnome 45 release is going to change up some dynamics with extensions, so unless you update your extension to support ESM, it is **NOT** going to work on Gnome 45. There is a blog post from the Gnome devs here that details briefly the reasons for this change and what we can do to help update extensions to the new release: https://blogs.gnome.org/shell-dev/2023/09/02/extensions-in-gnome-45/
> 
> tl;dr, GNOME Shell moved from GJS’ own custom imports system to standard JavaScript modules (ESM), which breaks every extension unless it's updated to support Gnome 45. Please let us know if your extension can be updated to support Gnome 45 (it's release should be within the month btw). I hope this can be updated in time, and thanks again for making a very useful little extension for Firefox 👍

This transition is quite a pain but I understand why they did this, it makes it easier for regular devs to build extensions.

**Comment by Rafostar** [comment]
I do not have GNOME 45 right now at hand. Can someone interested in this help testing #18 so it can be merged? Otherwise it will have to wait until I am able to do so.

**Comment by lorenzovecchietti** [comment]
I can help. Any precise test procedure I should follow?

At the moment I tried with Firefox, Brave and Clapper. Everything is working as expected.

(Tested on: GNOME Shell 45.0, openSUSE Tumbleweed, Kernel: 6.5.4-1)

**Comment by Scotty-Trees** [comment]
> I do not have GNOME 45 right now at hand. Can someone interested in this help testing #18 so it can be merged? Otherwise it will have to wait until I am able to do so.

Just curious do you use Ubuntu or Fedora? Both distros should be getting Gnome 45 released later this month, so we won't have to wait too much longer! Ubuntu 23.10 with Gnome 45 will release soon on October 12 and Fedora 39 will release on either Oct 17 for their first target date, or more realistically it'll be released the following week on Oct 24'th. Can't wait to see this extension get updated though because I'll be upgrading Fedora probably one day one, it's always hard to wait!

**Comment by Razer0123** [comment]
when will it be released to the store? installed fedora 39 and extension is not working
